components/dropDown/dropDown.js

```diff
@@ -0,0 +1,166 @@
+import React, { Component, PropTypes } from 'react';
+import Select from 'react-select';
+
+import { getClassNamesWithMods } from '../_helpers';
+import DropdownFilterOptionComponent from './dropdownFilterOptionComponent';
+
+/**
+ * DropDown component
+ */
+class DropDown extends Component {
+  constructor(props) {
+    super(props);
+
+    this.onChange = this.onChange.bind(this);
+    this.menuRenderer = this.menuRenderer.bind(this);
+  }
+
+  optionRef = (onOptionRef, isSelected) => ref => onOptionRef(ref, isSelected);
+
+  /**
+   * Overriding the internal method of react-select for fix autoscrolling
+   */
+  menuRenderer({
+      focusedOption,
+      instancePrefix,
+      onFocus,
+      onSelect,
+      optionClassName,
+      optionComponent,
+      optionRenderer,
+      options,
+      valueArray,
+      valueKey,
+      onOptionRef,
+    }) {
+    let Option = optionComponent;
+
+    return options.map((option, i) => {
+      let isSelected = valueArray && valueArray.indexOf(option) > -1;
+      let isFocused = option === focusedOption;
+      const classes = ['Select-option'];
+
+      optionClassName && classes.push(optionClassName);
+      isSelected && classes.push('is-selected');
+      isFocused && classes.push('is-focused');
```

Comment:
now, you can use object instead of array, to apply classes based on props:
```js
const mods = {
  'is-focused': isFocused,
  'is-selecte': isSelected
};

<Option classNames={getClassNamesWithMods('Select-option', mods)} />
```